src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +104,125 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
+                { TelemetryConstants.TokenSource, Convert.ToInt64(authenticationResult.AuthenticationResultMetadata.TokenSource) },
+                { TelemetryConstants.Duration, authenticationResult.AuthenticationResultMetadata.DurationTotalInMs },
+                { TelemetryConstants.DurationInCache, authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs },
+                { TelemetryConstants.DurationInHttp, authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs },
+                { TelemetryConstants.TokenType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.TokenType },
+                { TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds } 
+            };
+
+            if (authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue)
+            {
+                tags.Add(TelemetryConstants.RefreshOn, DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value));
+            }
+            tags.Add(TelemetryConstants.AssertionType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.AssertionType);
+            tags.Add(TelemetryConstants.Endpoint, AuthenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority.ToString());
+            tags.Add(TelemetryConstants.CacheLevel, (int)authenticationResult.AuthenticationResultMetadata.CacheLevel);
+
+            var resourceAndScopes = ParseScopesForTelemetry();
+
+            if (resourceAndScopes.Item1 != null)
+            {
+                tags.Add(TelemetryConstants.Resource, resourceAndScopes.Item1);
+            }
+            if (resourceAndScopes.Item2 != null)
+            {
+                tags.Add(TelemetryConstants.Scopes, resourceAndScopes.Item2);
+            }
+
+            ServiceBundle.PlatformProxy.OtelInstrumentation.LogActivity(tags);
+        }
+
+        private void LogMsalFailedTelemetryToOTel(Exception exception)
+        {
+            if (ServiceBundle.PlatformProxy.OtelInstrumentation == null)
+            {
+                return;
+            }   
+
+            ServiceBundle.PlatformProxy.OtelInstrumentation.LogActivityStatus(false);
+
+            Dictionary<string, object> tags = new Dictionary<string, object>
+            {
+                { TelemetryConstants.AssertionType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.AssertionType },
+                { TelemetryConstants.Endpoint, AuthenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority.ToString() },
+
+                { TelemetryConstants.ErrorMessage, exception.Message }
+            };
+
+            if (exception is MsalClientException clientException)
+            {
+                tags.Add(TelemetryConstants.ErrorCode, clientException.ErrorCode);
+                return;
+            }
+
+            if (exception is MsalServiceException serviceException)
```

Comment:
should we consider adding managedIdentityException?

Reply:
ManagedIdentityException is inherited from MsalServiceException so will fall in this category.